contrib/filter-repo-demos/timeshift.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import subprocess
+import datetime
+import sys
+try:
+  import git_filter_repo as fr
+except ImportError:
+  raise SystemExit("Error: Couldn't find git_filter_repo.py.  Did you forget to make a symlink to git-filter-repo named git_filter_repo.py or did you forget to put the latter in your PYTHONPATH?")
+
+parser = argparse.ArgumentParser(
+          description='Offset a set of or all commits by n seconds, can be used to timeshift a repository')
+parser.add_argument('--offset', type=int,
+        help=("Offset in seconds"))
+parser.add_argument('--from-commit', type=str,
+        help=("Commit to start from"))
+parser.add_argument('--to-commit', type=str,
+        help=("Commit to finish at"))
```

Comment:
The implementation of --from-commit and --to-commit here will work for linear histories and some very special non-linear ones but is in general quite broken to rely on the order of commits as output by fast-export rather than checking for actual inclusion of commits inside a history range.

Reply:
Yeah, good point - hadn't thought about non linear history repos, the repo I wrote this for is small and simple with a single master branch and timeline and the tool worked perfectly timeshifting groups of commits since the dev machine at the time had incorrect date/time